docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,289 @@
+    - [ ] Delegate call
+        - [ ] If the opcode modifies the storage of the account currently executing it, verify that only the account that is delegating execution is the one that has its storage modified.
+        - [ ] If the opcode modifies the balance of the account currently executing it, verify that only the account that is delegating execution is the one that has its balance modified.
+        - [ ] If the opcode modifies the code of the account currently executing it, verify that only the account that is delegating execution is the one that has its code modified.
+    - [ ] Code call
+    - [ ] Initcode
+        - [ ] Verify opcode behaves as expected when running during the initcode phase of contract creation
+            - [ ] Initcode of a contract creating transaction.
+            - [ ] Initcode of a contract creating opcode (including itself if opcode creates a contract).
+    - [ ] Transaction context
+        - [ ] If opcode changes behavior depending on particular transaction properties, test using multiple values for each property.
+    - [ ] Block context
+        - [ ] If opcode changes behavior depending on particular block properties, test using multiple values for each property.
+    - [ ] EOF Container Context
+        - [ ] If opcode changes behavior depending on particular EOF container properties, test using multiple values for each property.
+- [ ] Return data
+    - [ ] Verify proper return data buffer modification if the opcode is meant to interact with it, otherwise verify that the return data buffer is unnaffected
+        - [ ] At current call context.
+        - [ ] At parent call context.
+- [ ] Gas usage
+    - [ ] Normal operation gas usage
+        - [ ] Verify gas affectation of each parameter value consumed by the opcode.
+    - [ ] Memory expansion
+        - [ ] Verify that the memory expansion correctly follows the gas calculation
+    - [ ] Out-of-gas during opcode execution
+        - [ ] Verify that attempting to execute the opcode when gas available is 1 less than the required gas results in exeptional abort.
+    - [ ] Out-of-gas during memory expansion
+        - [ ] Verify that the expansion of memory can result in out-of-gas exeptional abort.
+- [ ] Terminating opcode
+    - [ ] If an opcode is terminating, meaning it results in the current call context to end execution, test the following scenarios
+        - [ ] Top-level call termination
+        - [ ] Sub-level call termination
+        - [ ] Initcode termination
+- [ ] Out-of-bounds checks
+    - [ ] Verify if the opcode has out-of-bounds conditions in its parameters and verify:
+        - [ ] Max value for each parameter
+        - [ ] Max value + 1 for each parameter
+- [ ] Exeptional Abort
+    - [ ] Verify behavior that is supposed to cause an exeptional abort, other than stack over or underflow, or out-of-gas errors.
```

Comment:
This is often not clear when I read EIPs what an "exceptional abort" means. Usually this means: consume all gas. But in other contexts (REVERT) the call frame is aborted/reverted, but any gas left in the current frame is returned to the caller (the "nice" option :)). Often have to go back to EIP author to figure out what is meant here.

Reply:
This is a nice observation. I think REVERT is more of a case of being a "terminating opcode", because it doesn't consume all gas, and is an expected abort rather than exceptional.

Nonetheless I added another bullet-point to the "Terminating opcode" section to account for terminating opcodes that rollback the call frame.